src/NuGet.Core/NuGet.Commands/PublicAPI/net472/PublicAPI.Unshipped.txt

```diff
@@ -0,0 +1,12 @@
+NuGet.Commands.CLILanguageOverrider
```

Comment:
Can we avoid adding a new public API in the shared APIs? 

IMO, we have a lot of abstractions here for something that should be fairly straightforward.

Reply:
I tried (see comment CLILanguageOverrider class), but it's the only way to share this logic between NuGet.CommandLine (nuget.exe) and NuGet.CommandLine.XPlat (dotnet.exe) projects 

If I am allowed to to create a new project or, add project references to CLI projects, then, I can remove the public API.

Reply:
New assemblies aren't free and come with paperwork. I'd suggest not adding them.
IMO we should just duplicate the code. 

I think the CLI override code is overly complicated as it. After all, it's just a few env vars and parsing, we shouldn't need a complicated logic for that.

Reply:
Even if a new assembly was created, the way you make it available to the existing projects (the ones you add a ProjectReference to) is with a public API in the new assembly. Therefore, it doesn't even solve the problem, it will still be a new public API.

By comparison, the dotnet CLI's code to override the language is here: https://github.com/dotnet/sdk/blob/a30e465a2e2ea4e2550f319a2dc088daaafe5649/src/Cli/dotnet/UILanguageOverride.cs
And the code for `dotnet new` is here: https://github.com/dotnet/sdk/blob/8ff664e92c7faa95cd1146cc1029caa7bb24e6aa/src/Cli/dotnet-new3/HostFactory.cs#L62-L112

They're in the same repo, but they don't share code.  They're implemented in 80 lines each, and in my opinion easy to understand (you can read the methon top-down, and follow what's going on). Whereas the two classes added to NuGet.Commands in this PR total around 170 lines, and use two `Func<>`'s, which make it significantly more difficult to understand via reading, since program flow is no longer "top down". Plus there's a bunch of memory allocations which is going to have to run every single time the app starts up, before it even starts doing the business logic of whatever command the customer tried to run.

While it might be ideal to have a single implementation that is reused in multiple places, I also think the tradeoff is not good and that we'd probably be better off duplicating the code in the two projects, and making them specific to each, and easy to read like in the dotnet/sdk repo.

Reply:
OK, I will create another separate implementation. Thanks all!